modelator/tests/integration/main.rs

```diff
@@ -2,19 +2,117 @@
 // https://matklad.github.io/2021/02/27/delete-cargo-integration-tests.html
 
 use modelator::artifact::{JsonTrace, TlaFile};
+use modelator::test_util::NumberSystem;
+use modelator::{ActionHandler, EventRunner, EventStream, StateHandler};
 use modelator::{CliOptions, CliStatus, Error, ModelChecker, ModelCheckerOptions, Options};
 use once_cell::sync::Lazy;
-use serde_json::{json, Value as JsonValue};
+use serde::Deserialize;
+use serde_json::Value as JsonValue;
 use std::path::{Path, PathBuf};
 use std::sync::Mutex;
 
+#[derive(Default, Debug, PartialEq)]
+struct Numbers {
+    a: i64,
+    b: i64,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+struct A {
+    a: u64,
+}
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+struct B {
+    b: u64,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+enum Action {
+    None,
+    IncreaseA,
+    IncreaseB,
+}
+
+impl StateHandler<A> for NumberSystem {
+    fn init(&mut self, state: A) {
+        self.a = state.a
+    }
+    fn read(&self) -> A {
+        A { a: self.a }
+    }
+}
+impl StateHandler<B> for NumberSystem {
+    fn init(&mut self, state: B) {
+        self.b = state.b
+    }
+    fn read(&self) -> B {
+        B { b: self.b }
+    }
+}
+
+impl ActionHandler<Action> for NumberSystem {
```

Comment:
Why `Action` is generic type and `Outcome` is associated type? `Action` does not have trait bounds. We can make `Action` as an associated type too, no?